app/src/main/java/org/mozilla/fenix/perf/ProfilerStopDialogFragment.kt

```diff
@@ -0,0 +1,265 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import androidx.fragment.app.DialogFragment
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.perf.ProfilerUtils.handleProfileSave
+import java.io.FileOutputStream
+
+/**
+ * Dialogue to stop the Gecko profiler without using ADB.
+ */
+class ProfilerStopDialogFragment : DialogFragment() {
+
+    private lateinit var profileSaveLauncher: ActivityResultLauncher<Intent>
+    private lateinit var profileArray: ByteArray
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        createFileSaverLauncher()
+        return ComposeView(requireContext()).apply {
+            setContent {
+                StopProfilerCard()
+            }
+        }
+    }
+
+    @Composable
+    private fun StopProfilerCard() {
+
+        val viewStateObserver = remember { mutableStateOf(CardState.StopDialogState) }
+        val saveProfileTypeObserver = remember { mutableStateOf("") }
+
+        Dialog(onDismissRequest = { this@ProfilerStopDialogFragment.dismiss() }) {
+            when (viewStateObserver.value) {
+                CardState.StopDialogState -> {
+                    StopCard(viewStateObserver, saveProfileTypeObserver)
+                }
+                CardState.UrlWarningState -> {
+                    UrlWarningCard(viewStateObserver, saveProfileTypeObserver)
+                }
+                CardState.WaitForProfilerState -> {
+                    WaitForProfilerDialog("Gathering the profile")
```

Comment:
issue (blocking): when the profiler is saving, I can click outside of the waiting dialog to dismiss it. Eventually, we get a callback that access `context` but the context is dead so the app crashes.

suggestion: don't allow the stop profiler waiting dialog to be dismissed (e.g. I also haven't tested the back button's behavior which might have the same effect)

Reply:
Ah yes, both back button and tapping outside the window would dismiss it. Good point